src/sql/parts/modelComponents/inputbox.component.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import 'vs/css!./flexContainer';
+
+import { Component, Input, Inject, ChangeDetectorRef, forwardRef, ComponentFactoryResolver,
+	ViewChild, ViewChildren, ElementRef, Injector, OnDestroy, QueryList, AfterViewInit
+} from '@angular/core';
+
+import * as sqlops from 'sqlops';
+import Event, { Emitter } from 'vs/base/common/event';
+
+import { ComponentBase } from 'sql/parts/modelComponents/componentBase';
+import { IComponent, IComponentDescriptor, IModelStore, ComponentEventType } from 'sql/parts/modelComponents/interfaces';
+import { InputBox, IInputOptions } from 'vs/base/browser/ui/inputbox/inputBox';
+import { CommonServiceInterface } from 'sql/services/common/commonServiceInterface.service';
+import { attachInputBoxStyler, attachListStyler } from 'vs/platform/theme/common/styler';
+
+@Component({
+	selector: 'inputBox',
+	template: `
+		<div #input style="width: 100%"></div>
+		<div class="inputBoxComponent" style="position: absolute; height: 100%; width: 100%; margin: 10px 0px 10px 0px;">
+			<span style="margin-left: 10px; display: inline-block;">
+				<div #input style="width: 100%"></div>
```

Comment:
Why have 2 divs with `#input` as their selector? The `@ViewChild` reference will only detect the one, right? Shouldn't you have a different name for the 2nd? I don't understand how this works.